### PR TITLE
chore: put `__tests__/.eslintrc.json` in root's config's `override`

### PR DESCRIPTION
**Pull request by nschonni**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] New rule
- [ ] Changes an existing rule
- [ ] Add autofixing to a rule
- [x] Other, please explain:

**What changes did you make? (Give an overview)**
Merged up the nested ESLint config to an override in the root one

### CODE REVIEW

**Comment by xjamundx** [comment]
Tell me more about why this is better. The old way made sense to me. This also seems fine. Just wondering if there's a big benefit.

**Comment by nschonni** [comment]
Yeah, either way is fine. Using the single file, you get to see all the rules and overrides in a single location when you want to add/remove plug-ins

File: .eslintrc.json [hunk 1]
```diff
@@ -23,5 +23,19 @@
...
+  },
+  "overrides": [
+    {
+      "files": ["__tests__/*.js"],
+      "plugins": ["jest"],
+      "extends": [
+        "eslint:recommended",
+        "plugin:jest/recommended",
+        "plugin:prettier/recommended"
+      ],
```
Comment (MichaelDeBoey) [inline .eslintrc.json:35]: We should keep this as `__tests__/.eslintrc.json` was

```suggestion
      "extends": ["plugin:jest/recommended"],
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (nschonni) [reply .eslintrc.json:35]: AFAIK, overrides wipes out the whole property, that's why the full set of extends is required

**Comment by voxpelli** [comment]
Something like this will need to be done when we adapt to flat config anyway, but I think it's better to focus on that rather than do this now that flat configs are the future

**Comment by brettz9** [comment]
And FWIW, FWICT, with flat configs at least, the "overrides" equivalent does not wipe out the whole property.

**Pull request closed without merging**